code/modules/mod/mod_activation.dm

```diff
@@ -0,0 +1,149 @@
+/obj/item/mod/control/proc/choose_deploy(mob/user)
+	if(!LAZYLEN(mod_parts))
+		return
+	if(isAI(user))
+		to_chat(user, "<span class='warning'>You cannot operate this!</span>")
+		return
+	var/list/display_names = list()
+	var/list/items = list()
+	for(var/i in 1 to length(mod_parts))
+		var/obj/item/piece = mod_parts[i]
+		display_names[piece.name] = REF(piece)
+		var/image/piece_image = image(icon = piece.icon, icon_state = piece.icon_state)
+		items += list(piece.name = piece_image)
+	var/pick = show_radial_menu(user, src, items, custom_check = FALSE, require_near = TRUE)
+	if(!pick)
+		return
+	var/part_reference = display_names[pick]
+	var/obj/item/part = locate(part_reference) in mod_parts
+	if(!istype(part) || user.incapacitated())
+		return
+	if(active || activating)
+		to_chat(user, "<span class='warning'>ERROR: Suit activated. Deactivate before further action.</span>")
+		playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE)
+		return
+	if(part.loc == src)
+		deploy(user, part)
+	else
+		conceal(user, part)
+
+/obj/item/mod/control/proc/deploy(mob/user, part)
+	var/obj/item/piece = part
+	if(piece == gauntlets && wearer.gloves)
+		gauntlets.overslot = wearer.gloves
+		wearer.transferItemToLoc(gauntlets.overslot, gauntlets, TRUE)
+	if(piece == boots && wearer.shoes)
+		boots.overslot = wearer.shoes
+		wearer.transferItemToLoc(boots.overslot, boots, TRUE)
+	if(wearer.equip_to_slot_if_possible(piece,piece.slot_flags, qdel_on_fail = FALSE, disable_warning = TRUE))
+		user.visible_message("<span class='notice'>[wearer]'s [piece] deploy[piece.p_s()] with a mechanical hiss.</span>",
+			"<span class='notice'>[piece] deploy[piece.p_s()] with a mechanical hiss.</span>",
+			"<span class='hear'>You hear a mechanical hiss.</span>")
+		playsound(src, 'sound/mecha/mechmove03.ogg', 25, TRUE)
+		ADD_TRAIT(piece, TRAIT_NODROP, MOD_TRAIT)
+	else if(piece.loc != src)
+		to_chat(user, "<span class='warning'>ERROR: [piece] [piece.p_are()] already deployed.</span>")
+		playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE)
+	else
+		to_chat(user, "<span class='warning'>ERROR: Bodypart clothed.</span>")
```

Comment:
maybe this should try to force the item covering that slot onto the ground if it doesn't have nodrop or something?

also, this should probably produce a different message if you physically can't wear something in that slot, due to being a golem or something

Reply:
just atke it off